modules/examiner/php/editexaminer.class.inc

```diff
@@ -381,35 +378,26 @@ class EditExaminer extends \NDB_Form
      */
     function _validateEditExaminer($values)
     {
-        $DB     = \Database::singleton();
         $errors = array();
 
-        // check that there is both a status and a date (neither can be null)
+        $instruments = $this->getCertificationInstruments();
         foreach ($values['certStatus'] as $instrumentID => $certificationStatus) {
-            if (empty($certificationStatus)
-                || empty($values['date_cert'][$instrumentID])
-            ) {
-                if (($certificationStatus == "certified")
-                    && empty($values['date_cert'][$instrumentID])
-                ) {
-                    $errors['certStatus[' . $instrumentID .']'] = 'Both certification
-                        status and date must be filled out';
-                }
+
+            // mandatory to choose an option for the Certification Status.
+            if (empty($certificationStatus)) {
+                $errors['certStatus[' . $instrumentID .']'] = 'Please choose
+                    a Certification Status for the Instrument: '
+                    . $instruments[$instrumentID];
             }
-        }
 
-        // check if previously recorded certification are all present
-        // (can not delete, only change status)
-        $rows = $DB->pselect(
-            "SELECT c.testID
-              FROM certification c
-              WHERE c.examinerID=:EID",
-            array('EID' => $this->identifier)
-        );
-        foreach ($rows as $row) {
-            if ($values['certStatus'][$row['testID']] == "") {
-                $errors['certStatus['.$row['testID'].']']
-                    = 'You can not delete a status';
```

Comment:
Removing this allows users to change status from certified > anything else. Not sure if this is intended behaviour?

Reply:
Hi @jesscall, 

What I understand from this removed code is that is not allowing to change from `ANY state different from BLANK(a sort of null) to BLANK.` Not precisely preventing to change from `CERTIFIED TO ANYTHING ELSE.`

If that the case, is already covered in the new lines 386 to 390:
```
            // mandatory to choose an option for the Certification Status.
	    if (empty($certificationStatus)) {
                $errors['certStatus[' . $instrumentID .']'] = 'Please choose
                a Certification Status for the Instrument: '
                . $instruments[$instrumentID];
```

Reply:
@racostas as far as i understand it, these statements are not equivalent. Your new code will prevent saving the form with any of the certifications empty !! which means you would not be able to partially certify an examiner (that's not good)

the old code looked in the existing certification table and made sure that IF you were previously certified for a test, you cannot go back to `""` which means you cannot go from a certification to nothing, but you can still save with empty certifications if they were never set beforehand

Reply:
Hi @ridz1208, 

Yes, this is totally the case. Completely agree. The previous code if I don't remember bad was preventing also that null values were saved even for first time in the database (maybe not intended). That could be not the expected behaviour and that's why I put a note on the description of this PR regarding the N/A values and null values.

I limited my fixing so far (waiting for comment or discussion on this other aspect) to grant just the same functionalities available before. The available certification status are (without taking N/A or null into account): Not Certified, In Training and Certified. N/A (or null, I'm more inclined to N/A) will be useful in a case we don't know the state, and could be valuable in my opinion.

I'm not completely sure if we want to prevent going from previously certificated to N/A in the case we decide to include N/A or null as an option. What if the "data entry" person assign one of the available status: Not Certified, In Training and Certified and after saving he/she realises made a mistake and assigned a status to the examiner but when corroborating realises didn't have information about it? In this case the user will not be able to rollback to N/A and will have to stick to any of the other three status (when not exactly the reality from the business perspective).

I'm very willing to implement the solution we decide fix best the user case.

Reply:
@driusan do you know the reason why we never allowed unsetting the certification status ?

Reply:
no, maybe @zaliqarosli does?

Reply:
I think this predates Zaliqa by alot. I'll message Leigh

Reply:
looks like that section was added in this PR https://github.com/aces/Loris/pull/3359/files. I commented on it saying that I liked the added check, but I don't remember the reason I was for it